### PR TITLE
fix bug where gizmos does not show up on the correct position

### PR DESCRIPTION
**Pull request by alonss0**

### CODE REVIEW

**Comment by MGH-Automator** [comment]
Task linked: [CU-3tmdjkd Fix: Gizmo not properly aligned with object when the object parents are rotated and scaled in a certain way](https://app.clickup.com/t/3tmdjkd)

File: Assets/Scripts/SceneState/DclTransformComponent.cs [hunk 1]
```diff
@@ -20,67 +20,112 @@ public DclTransformComponent(DclComponent c) : base(c.NameInCode, c.NameInCode)
...
+                localMatrix.SetTRS(Position.Value, Rotation.Value, Scale.Value);
+
+                var globalMatrix = parentMatrix * localMatrix;
 
-                return globalPosition;
+                return globalMatrix;
+            }
+        }
+
+        public Quaternion GlobalRotation
```
Comment (cblech) [inline Assets/Scripts/SceneState/DclTransformComponent.cs]: Please move the GlobalRotation under the GlobalPosition, so we have consistent sorting

File: Assets/Scripts/SceneState/DclTransformComponent.cs [hunk 2]
```diff
@@ -20,67 +20,112 @@ public DclTransformComponent(DclComponent c) : base(c.NameInCode, c.NameInCode)
...
+            {
+                return GlobalTransformMatrix.rotation;
             }
         }
+
         public Vector3 GlobalPosition
         {
             get
             {
                 if (Entity.Parent == null)
```
Comment (cblech) [inline Assets/Scripts/SceneState/DclTransformComponent.cs]: You don't need this if-branch. You are doing the same thing on both branches

File: Assets/Scripts/SceneState/DclTransformComponent.cs [hunk 3]
```diff
@@ -20,67 +20,112 @@ public DclTransformComponent(DclComponent c) : base(c.NameInCode, c.NameInCode)
...
+                return globalMatrix;
             }
         }
 
-        public Quaternion GlobalRotation
+        public Vector3 GlobalFixedPosition
         {
             get
             {
                 if (Entity.Parent == null)
```
Comment (cblech) [inline Assets/Scripts/SceneState/DclTransformComponent.cs]: You don't need this if-branch. Same as above

**Review by cblech — CHANGES_REQUESTED** [review]

**Review by cblech — APPROVED** [review]

**Pull request merged**